web_monitoring/cli.py

```diff
@@ -0,0 +1,55 @@
+# Command Line Interface
+# All the functionality is implemented in this module. See scripts/ for
+# executables.
+from docopt import docopt
+import hashlib
+import io
+import lxml.html
+import requests
+from tqdm import tqdm
+from web_monitoring import internetarchive as ia
+from web_monitoring import db
+
+
+# These functions lump together library code into monolithic operations for the
+# CLI. They also print. To access this functionality programmatically, it is
+# better to use the underlying library code.
+
+
+def import_ia(url, agency, site):
+    print('obtaining versions list from Internet Archive...')
+    versions = ia.list_versions(url)
+    # Collect all the results and POST them as a unit (rather than streaming).
+    formatted_versions = []
+    for dt, uri in tqdm(versions, desc='formatting versions'):
+        res = requests.get(uri)
+        assert res.ok
+        version_hash = hashlib.sha256(res.content)
+        content_as_file = io.StringIO(res.content.decode(errors='ignore'))
```

Comment:
Don't worry, the `version_hash` is performed on raw undecoded bytes. This `content_as_file` is just a temporary var from which I am parsing the 'title' tag to auto-extract the Page title (to be used by humans, not machines). I have decode from bytes to string before I can parse XML. I think ignoring errors is fine here because (1) fidelity is not critical here and (2) errors seem rare.